vulkan_generator/vulkan_gen/parser/types.py

```diff
@@ -0,0 +1,57 @@
+# Copyright (C) 2022 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This module contains the Vulkan meta data definitions to define Vulkan types and functions"""
+
+from dataclasses import dataclass
+from enum import Enum
+
+
+# Melih TODO: Complete
+class VulkanTypeCategories(Enum):
+    """Enum to define different kinds of Vulkan types"""
+    VULKAN_HANDLE = 1
+    VULKAN_HANDLE_ALIAS = 2
+    VULKAN_ENUM = 3
+    VULKAN_STRUCT = 4
+
+
+@dataclass
+class VulkanType:
+    """Base class for a Vulkan Type. All the other Types should inherit from this class"""
```

Comment:
In terms of Python category might be redundant, I will think of it again. I assume we can easily use isInstanceof or something similar.

But for the polymorphism,  I thought if should we have or not and I decided to have at the last minute.

One of the reasons is that, Vulkan XML does not separate between a type and alias. So, when we parse for example a struct alias, it will be under the structs tag but we will not have any idea about what the actual struct is, it will only be defining an alias. It is kinda useful to have it so that we can just return VulkanType and decide later what to do with it. It kinda makes the code simpler(at least for now). Please let me know if you have any objection.